### PR TITLE
Support Slice on DML backend

### PR DESCRIPTION
**Pull request by fujunwei**

### CODE REVIEW

File: src/webnn_native/dml/GraphDML.cpp [hunk 1]
```diff
@@ -1073,6 +1073,48 @@ namespace webnn_native { namespace dml {
         return {};
     }
 
+#define SLICE_ONE_AXIS(axis, index)                                                      \
+    inputWindwOffsets[axis] =                                                            \
```
Comment (huningxin) [inline src/webnn_native/dml/GraphDML.cpp:1077]: typo of `inputWindowOffsets`?

File: src/webnn_native/dml/GraphDML.cpp [hunk 2]
```diff
@@ -1073,6 +1073,48 @@ namespace webnn_native { namespace dml {
...
+        starts[index] < 0 ? (starts[index] + inputDims[axis]) : starts[index];           \
+    inputWindwSizes[axis] =                                                              \
```
Comment (huningxin) [inline src/webnn_native/dml/GraphDML.cpp:1079]: ditto

File: src/webnn_native/dml/GraphDML.cpp [hunk 3]
```diff
@@ -1073,6 +1073,48 @@ namespace webnn_native { namespace dml {
...
+
+    MaybeError Graph::AddSlice(const op::Slice* slice) {
+        DAWN_ASSERT(slice->Inputs().size() == 1);
+        const OperandBase* inputOperand = slice->Inputs()[0].Get();
+        DAWN_ASSERT(mExpression.find(inputOperand) != mExpression.end());
+        ::dml::Expression input = mExpression.at(inputOperand);
+        ::dml::TensorDimensions inputDims = input.GetOutputDesc().sizes;
+
+        // dml::Span just holds the refernces, need a variable to hold the memory.
+        std::vector<uint32_t> inputWindwOffsets(inputDims.size(), 0);
```
Comment (huningxin) [inline src/webnn_native/dml/GraphDML.cpp:1092]: ditto

File: src/webnn_native/dml/GraphDML.cpp [hunk 4]
```diff
@@ -1073,6 +1073,48 @@ namespace webnn_native { namespace dml {
...
+        std::vector<uint32_t> inputWindwSizes(inputDims);
```
Comment (huningxin) [inline src/webnn_native/dml/GraphDML.cpp:1093]: ditto

File: src/webnn_native/dml/GraphDML.cpp [hunk 5]
```diff
@@ -1073,6 +1073,48 @@ namespace webnn_native { namespace dml {
...
+            }
+        } else {
+            for (size_t i = 0; i < axes.size(); ++i) {
+                if (axes[i] < 0) {
+                    axes[i] = inputDims.size() + axes[i];
+                }
+                SLICE_ONE_AXIS(axes[i], i);
+            }
+        }
+        std::vector<int32_t> inputWindwStrides(inputDims.size(), 1);
```
Comment (huningxin) [inline src/webnn_native/dml/GraphDML.cpp:1109]: ditto

**Comment by fujunwei** [comment]
Done, thanks.

**Comment by fujunwei** [comment]
Let's merge it first, please let me know if you have any other comments.

**Pull request merged**